lib/zig.h

```diff
@@ -180,10 +180,16 @@ typedef char bool;
 #define zig_export(sig, symbol, name) __asm(name " = " symbol)
 #endif
 
+#if zig_has_builtin(trap)
+#define zig_trap() __builtin_trap()
+#elif defined(__i386__) || defined(__x86_64__)
+#define zig_trap() __asm__ volatile("ud2");
+#else
+#define zig_trap() raise(SIGILL)
```

Comment:
```suggestion
#define zig_trap() raise(SIGTRAP)
```

Reply:
The references I quoted and linked in [this comment](https://github.com/ziglang/zig/pull/14782#discussion_r1124885845) might be of interest.

> The SIGTRAP signal is sent to a process when an exception (or trap) occurs: a condition that **a debugger has requested to be informed of**

(this is used for `debugtrap` below)

and then SIGILL (used for `trap`):

> The SIGILL signal is sent to a process when it attempts to execute an **illegal, malformed, unknown, or privileged instruction**

Because they talk about debuggers in SIGTRAP. I know the signal names themselves make this seem like an odd choice but that's why I did it this way.